src/type_checker/checker.rs

```diff
@@ -394,7 +394,7 @@ impl<B: Backend> TypeChecker<B> {
                         .expect("expected a value");
 
                     if let Some(tykind) = &tykind {
-                        if !tykind.same_as(&item_typ.typ) {
+                        if !tykind.compare_with(&item_typ.typ,true) {
```

Comment:
so `same_as` is only used in a single place, but it doesn't make sense to me. Why wouldn't we be able to declare an array with a mix of constants and non-constants? We might want to get rid of the `same_as` function altogether

Reply:
actually, I'm backtracking here, I think for array declaration we'll want to have a flag that says we don't even care about the constant in a Field, maybe the flag could be called `ignore_constants`